unified-scheduler-logic/src/lib.rs

```diff
@@ -1,20 +1,1299 @@
-use solana_sdk::transaction::SanitizedTransaction;
+#![allow(rustdoc::private_intra_doc_links)]
+//! The task (transaction) scheduling code for the unified scheduler
+//!
+//! ### High-level API and design
+//!
+//! The most important type is [`SchedulingStateMachine`]. It takes new tasks (= transactions) and
+//! may return back them if runnable via
+//! [`::schedule_task()`](SchedulingStateMachine::schedule_task) while maintaining the account
+//! readonly/writable lock rules. Those returned runnable tasks are guaranteed to be safe to
+//! execute in parallel. Lastly, `SchedulingStateMachine` should be notified about the completion
+//! of the exeuction via [`::deschedule_task()`](SchedulingStateMachine::deschedule_task), so that
```

Comment:
kitten dies. :crying_cat_face: well, it's really bad condition, i haven't thought about  in depth. it will easily lead to dead locks.

well, i wonder this edge case should be called out in this summary doc.. fyi, I explicit mentioned this case in the `deschedule_task()` doc comment: 250edde6dde10899ffb7c05d1b143fca4b0f1705